content/blog/recipes-1-1-0/index.Rmd

```diff
@@ -0,0 +1,239 @@
+
+```{r}
+recipes_ptype(rec)
+```
+
+Note that recipes created before version 1.1.0 don't contain any ptype information, and will not undergo checking. Rerunning the code to specify the recipe will add ptype information to the recipe.
+
+## Input checking in recipe()
+
+Every recipe you create start with a call to `recipe()`. We have relaxed the requirements of data frames, while increasing the feedback when something goes wrong.
+
+The data was previously passed through `model.frame()` inside the recipe, which restricted what could be handled. Previously prohibited input included data frames with list-columns or [sf](https://r-spatial.github.io/sf/) data frames. Both of these are now supported, as long as they are a `data.frame` object.
+
+```{r}
+data_listcolumn <- tibble(
+  y = 1:4,
+  x = list(1:3, 4:6, 3:1, 1:10)
+)
+
+recipe(y ~ ., data = data_listcolumn)
+```
+
+```{r}
+library(sf)
+pathshp <- system.file("shape/nc.shp", package = "sf")
+data_sf <- st_read(pathshp, quiet = TRUE)
+
+recipe(AREA ~ ., data = data_sf)
+```
+
+We are excited to see what people can do with these new options.
+
+Another way to specify a recipe is to use `add_role()` and `update_role()`. But if you are not careful, you can end up in situations where the same variable is labeled as both the outcome and predictor.
+
+```{r}
+#| error: true
+# didn't use to throw a warning
+recipe(mtcars) |>
+  update_role(everything(), new_role = "predictor") |>
+  add_role("mpg", new_role = "outcome")
+```
+
+This specific problem can be dealt with using `update_role()` instead of `add_role()`.
+
+```{r}
+recipe(mtcars) |>
+  update_role(everything(), new_role = "predictor") |>
+  update_role("mpg", new_role = "outcome")
+```
+
+## Long formulas in recipe()
+
+Related to the changes we saw above, we now fully support very long formulas without hitting a `C stack usage` error.
+
+```{r}
+data_wide <- matrix(1:10000, ncol = 10000)
+data_wide <- as.data.frame(data_wide)
+names(data_wide) <- c(paste0("x", 1:10000))
+
+long_formula <- as.formula(paste("~ ", paste(names(data_wide), collapse = " + ")))
+
+recipe(long_formula, data_wide)
+```
+
+## Better error for misspelled argument names
+
+If you have used recipes long enough you are very likely to have run into the following error:
+
+``` r
+recipe(mpg ~ ., data = mtcars) |>
+  step_pca(all_numeric_predictors(), number = 4) |>
+  prep()
+#> Error in `step_pca()`:
+#> Caused by error in `prep()`:
+#> ! Can't rename variables in this context.
+```
+
+and the first time you saw it, it didn't make much sense. Hopefully, you figured out that [step_pca()](https://recipes.tidymodels.org/reference/step_pca.html) doesn't have a `number` argument, and instead uses `num_comp` to determine the number of principal components to return. This confusion will be a thing of the past as we now include this improved error message:
+
+```{r}
+#| error: true
+recipe(mpg ~ ., data = mtcars) |>
+  step_pca(all_numeric_predictors(), number = 4) |>
+  prep()
```

Comment:
Noting that this error message has a pluralization issue.

> The following argument was specified but do not exist:

Should be

> The following argument was specified but does not exist:

Reply:
added issue here https://github.com/tidymodels/recipes/issues/1345